tools/unit_tests.ts

```diff
@@ -0,0 +1,119 @@
+import { args, exit, run } from "deno";
+
+function extractNumber(pattern: RegExp, str: string) {
+  const matches = pattern.exec(str);
+  if (matches == null || pattern.exec(str) != null) {
+    return;
+  }
+  return Number(matches[1]);
+}
+
+function parseUnitTestOutput(
+  output: string
+): [string | undefined, string | undefined] {
+  let expected;
+  let actual;
+  let result;
+  for (const line of output.split("\n")) {
+    if (expected == null) {
+      // expect "running 30 tests"
+      expected = extractNumber(/running (\d+) tests/g, line);
+    } else if (line.includes("test result:")) {
+      result = line;
+    }
+    console.log(line);
+  }
+  // Check that the number of expected tests equals what was reported at the
+  // bottom.
+  if (result) {
+    // result should be a string like this:
+    // "test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; ..."
+    actual = extractNumber(/(\d+) passed/g, result);
+  }
+  return [actual, expected];
+}
+
+interface ReadResult {
+  eof: boolean;
+  nread: number;
+}
+
+interface Reader {
+  read(p: Uint8Array): Promise<ReadResult>;
+}
+
+async function readAll(reader: Reader) {
+  let result: Uint8Array;
+  while (true) {
+    const buffer = new Uint8Array(4096);
+    const { eof, nread } = await reader.read(buffer);
+    if (result) {
+      const tmp = result;
+      result = new Uint8Array(tmp.length + nread);
+      result.set(tmp);
+      result.set(buffer.slice(0, nread), tmp.length);
+    } else {
+      result = buffer.slice(0, nread);
+    }
+    if (eof) {
+      break;
+    }
+  }
+  return new TextDecoder("utf-8").decode(result);
+}
+
+async function runUnitTest2(args: string[]) {
+  const process = run({
+    args,
+    stdin: "null",
+    stdout: "piped"
+  });
+  const { code: errorCode } = await process.status();
+  const output = await readAll(process.stdout);
```

Comment:
Great work! I think we should have a more general purpose utility function which buffers the output. Something like `process.output(): Promise<Uint8Array>`
That should be done in a separate PR too. It can probably be ported from Golang:
https://github.com/golang/go/blob/048c9164a0c5572df18325e377473e7893dbfb07/src/os/exec/exec.go#L498-L520